plaidml2/edsl/edsl_test.py

```diff
@@ -98,10 +105,338 @@ def arg_max(I):
     O[x0, x2] >= cond(I[x0, x1, x2], Max[x0, x2], IX[x1])
     return as_uint(O, 32)
 
+def sum_over_axis(I):
+    M, N = TensorDims(2)
+    m, n = TensorIndexes(2)
+    I.bind_dims(M, N)
+    O = TensorOutput(N)
+    O[n] += I[m, n]; # contraction
+    return O
+
+def max_over_axis(I):
+    M, N = TensorDims(2)
+    m, n = TensorIndexes(2)
+    I.bind_dims(M, N)
+    O = TensorOutput(N)
+    O[n] >= I[m, n]
+    return O
+
+def matmul(A, B):
+    I, J, K =  TensorDims(3)
+    i, j, k = TensorIndexes(3)
+    A.bind_dims(I, K)
+    B.bind_dims(K, J)
+    C = TensorOutput(I, J)
+    C[i, j] += A[i, k] * B[k, j]
+    return C
+
+def global_min(I):
+    i, j, k = TensorIndexes(3)
+    Neg = -I
+    O_Neg = TensorOutput()
+    O_Neg[()] >= Neg[i, j, k]
```

Comment:
My memory might be bad, but I thought we had worked out that it was ok to do `O_Neg[]` instead of `O_Neg[()]`. If so, switch to the version without the extra parentheses (here and the other places it comes up).

Reply:
I get a syntax error with O_Neg[] version so sticking with O_Neg[()] for now.